Some rudimentary V4L2 drivers implement not even a single menu item in an
exposed menu, resulting in a segfault during the creation of the ControlInfo()
of the menu. This patch amends the code to throw an exception when trying to
create an empty menu, so that the menu can be ignored completely.

diff --git a/src/libcamera/v4l2_device.cpp b/src/libcamera/v4l2_device.cpp
index c17b323f8af0..57524d40f7a6 100644
--- a/src/libcamera/v4l2_device.cpp
+++ b/src/libcamera/v4l2_device.cpp
@@ -583,6 +583,16 @@ ControlInfo V4L2Device::v4l2MenuControlInfo(const struct v4l2_query_ext_ctrl &ct
 		indices.push_back(index);
 	}
 
+	/*
+	 * Bug: 167
+	 * Some UVC drivers are known to give return an empty menu control.
+	 * Populate the indicies with the default value only.
+	 */
+	if (indices.size() == 0) {
+		LOG(V4L2, Debug) << "Faulty menu control: " << ctrl.name;
+		indices.push_back(static_cast<int32_t>(ctrl.default_value));
+	}
+
 	return ControlInfo(indices,
 			   ControlValue(static_cast<int32_t>(ctrl.default_value)));
 }
